app/routes/audit/create-report-date.js

```diff
@@ -0,0 +1,89 @@
+const getClaimCountOverThreshold = require('../../services/data/audit/get-claim-count-over-threshold')
+
+const MIN_YEAR = 1900
+let validationErrors
+
+module.exports = function (router) {
+  router.get('/audit/create-report-date', function (req, res, next) {
+    authorisation.hasRoles(req, [applicationRoles.BAND_9, applicationRoles.CASEWORK_MANAGER_BAND_5])
+
+    getAuditConfig().then(function (result) {
+      const thresholdAmount = result.ThresholdAmount
+      addAuditSessionData(req, audit.SESSION.THRESHOLD_AMOUNT, thresholdAmount)
+      res.render('audit/create-report-date', {
+        thresholdAmount
+      })
+    })
+  })
+
+  router.post('/audit/create-report-date', function (req, res, next) {
+    authorisation.hasRoles(req, [applicationRoles.BAND_9, applicationRoles.CASEWORK_MANAGER_BAND_5])
+    validationErrors = {}
+    const auditReportStartDate = validateDate(
+      'auditReportStartDate',
+      req.body.auditReportStartDateDay,
+      req.body.auditReportStartDateMonth,
+      req.body.auditReportStartDateYear
+    )
+    const auditReportEndDate = validateDate(
+      'auditReportEndDate',
+      req.body.auditReportEndDateDay,
+      req.body.auditReportEndDateMonth,
+      req.body.auditReportEndDateYear
+    )
+    if (auditReportStartDate && auditReportEndDate && auditReportStartDate.isAfter(auditReportEndDate)) {
+      validationErrors.auditReportStartDate = ['The start date must be before the end date']
+    }
+    for (const field in validationErrors) {
+      if (Object.prototype.hasOwnProperty.call(validationErrors, field)) {
+        if (validationErrors[field].length > 0) {
+          return res.status(400).render('audit/create-report-date', {
+            query: req.body,
+            errors: validationErrors
+          })
+        }
+      }
+    }
+
+    addAuditSessionData(req, audit.SESSION.START_DATE, auditReportStartDate)
+    addAuditSessionData(req, audit.SESSION.END_DATE, auditReportEndDate)
+    const thresholdAmount = getAuditSessionData(req, audit.SESSION.THRESHOLD_AMOUNT)
+    getClaimCount(auditReportStartDate, auditReportEndDate).then(function (result) {
+      getClaimCountOverThreshold(auditReportStartDate, auditReportEndDate, thresholdAmount).then(function (result1) {
+        addAuditSessionData(req, audit.SESSION.CLAIM_COUNT, result[0].Count)
+        addAuditSessionData(req, audit.SESSION.CLAIM_COUNT_OVER_THRESHOLD, result1[0].Count)
+        return res.redirect('/audit/create-report-percent')
+      })
+    })
+  })
+}
+
+function validateDate (fieldName, day, month, year) {
+  const validationFieldName = validationFieldNames[fieldName] || 'Date'
+  if (day || month || year) {
+    const date = dateFormatter.build(day, month, year)
+    if (year >= MIN_YEAR && date.isValid()) {
+      if (dateFormatter.isFutureDate(date)) {
+        validationErrors[fieldName] = [validationFieldName + ' must be in the past']
+        return false
+      } else {
```

Comment:
All these `else` can go, just need to end the `if` and do the `return` after it. So:

```
      if (dateFormatter.isFutureDate(date)) {
        validationErrors[fieldName] = [validationFieldName + ' must be in the past']
        return false
      }

      return date
```

Just reads better.

Reply:
> All these `else` can go, just need to end the `if` and do the `return` after it. So:
> 
> ```
>       if (dateFormatter.isFutureDate(date)) {
>         validationErrors[fieldName] = [validationFieldName + ' must be in the past']
>         return false
>       }
> 
>       return date
> ```
> 
> Just reads better.

Updated